packages/orm/src/drivers/PostgresDriver.ts

```diff
@@ -379,31 +387,50 @@ async function batchUpdate(knex: Knex, meta: EntityMetadata<any>, entities: Enti
                   const maybeArray = c.serde instanceof EnumArrayFieldSerde ? "[]" : "";
                   return `?::${c.dbType}${maybeArray}`;
                 })
-                .join(",")})`,
+                .join(", ")}, ?::timestamp)`,
           )
           .join(",")}
-      ) AS data(${columns.map((c) => c.columnName).join(",")})
-      WHERE ${meta.tableName}.id = data.id
+      ) AS data(${columns.map((c) => c.columnName).join(",")},original_updated_at)
+      WHERE
+        ${meta.tableName}.id = data.id
+        AND date_trunc('milliseconds', ${meta.tableName}.updated_at) = data.original_updated_at
```

Comment:
The `date_trunc` is because PG has higher resolution than the JS dates we have, so need to round to milliseconds.